### DISCUSSION TITLE
Formatting Help

### DISCUSSION THREAD

**Discussion by DanielMartinGit** [nlohmann/json#3317] [Q&A]
So i'm currently working on a 2D game engine using your JSON library for serialising and deserialising scenes. So far I've managed to get it to loop through each entity and submit it's information to a JSON instance which is then input into the corresponding scene file. However, the information input to said scene file isn't in a correct JSON format. How would i go about formatting this with code so it goes from this: 

```
{
    "Entity ID: ": 1,
    "Entity Name: ": "Entity 1",
    "Entity Tag: ": "Entity"
}{
    "Entity ID: ": 0,
    "Entity Name: ": "Entity 0",
    "Entity Tag: ": "Entity"
}
```

to this:
```json
[{
    "Entity ID: ": 1,
    "Entity Name: ": "Entity 1",
    "Entity Tag: ": "Entity"
},{
    "Entity ID: ": 0,
    "Entity Name: ": "Entity 0",
    "Entity Tag: ": "Entity"
}]
```

Below is the code function I'm currently calling when I want the scene to be serialised:

```cpp
static void SerialiseEntity(entt::entity& entity, std::ofstream& sceneFile)
{
	nlohmann::json entityToSerialise =
	{
		{"Entity Name: ", Fuse::EntitySystem::GetComponent<Fuse::Entity>(entity).GetEntityName() },
		{"Entity ID: ", Fuse::EntitySystem::GetComponent<Fuse::Entity>(entity).GetEntityID() },
		{"Entity Tag: ", Fuse::EntitySystem::GetComponent<Fuse::Entity>(entity).GetEntityTag() }
	};

	sceneFile << std::setw(4) << entityToSerialise;
}
```

**Comment by gregmarr** [comment]
You would either need to manually write out the array support (`[` at the beginning, `,` between entities, `]` at the end), or pass a `json` object instead of an `ofstream` object and have the entity add their data to the `json` object.

**Reply by DanielMartinGit** [reply]
So the issue I found is, because I'm looping through the entities in the registry, every time that method is called it does write the necessary data, but it creates a whole new json object for each entity.
